apiserver/src/main/java/org/dependencytrack/tasks/maintenance/WorkflowMaintenanceTask.java

```diff
@@ -107,6 +121,25 @@ private Statistics informLocked(final Handle jdbiHandle) {
                 return;
             }
 
+            // Send BOM_PROCESSING_FAILED notifications for timed out failures.
+            final var notificationSubjectDao = jdbiHandle.attach(NotificationSubjectDao.class);
```

Comment:
Notifications should only be generated when `ConfigKey.TMP_DELAY_BOM_PROCESSED_NOTIFICATION` is `true`, and the failed steps are `WorkflowStep.VULN_ANALYSIS`.

If `TMP_DELAY_BOM_PROCESSED_NOTIFICATION` is `false` (default), then the `BOM_PROCESSING_FAILED` notification can not be sent after `BOM_PROCESSING` completed.

`TMP_DELAY_BOM_PROCESSED_NOTIFICATION` specifically delays the `BOM_PROCESSED` notification until after vulnerability analysis. If any other step (e.g. `REPO_META_ANALYSIS`, `METRICS_UPDATE`) fails, no notification should be sent.